src/core/profiling/endoscope_backend.h

```diff
@@ -0,0 +1,163 @@
+#include <grpc/support/sync.h>
+#include <grpc/support/time.h>
+#include <stdlib.h>
+
+#ifndef ENDOSCOPE_BACKEND_H_
+#define ENDOSCOPE_BACKEND_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Data types */
+
+#define ENDO_MUTEX_TYPE gpr_mu
+#define ENDO_INT64 gpr_int64
+#define ENDO_INDEX unsigned short
+#define ENDO_EMPTY (ENDO_INDEX)(-1)
+
+/* Public Macros */
+
+#define ENDOSCOPE_INIT(base) endoscope_init(base)
+#define ENDOSCOPE_DESTROY(base) endoscope_destroy(base)
+#define ENDOSCOPE_BEGIN(base, name) \
+        do { \
+          ENDO_INT64* temp_ = endoscope_begin(base, name, \
+              __FILE__, __LINE__, __PRETTY_FUNCTION__); \
+          *temp_ = ENDO_CYCLENOW(); \
+        } while (0)
+#define ENDOSCOPE_END(base, name) endoscope_end(base, name, ENDO_CYCLENOW())
+#define ENDOSCOPE_EVENT(base, name) endoscope_event(base, name, \
+            __FILE__, __LINE__, __PRETTY_FUNCTION__, ENDO_CYCLENOW())
+#define ENDOSCOPE_ERROR(base, name) endoscope_error(base, name, \
+            __FILE__, __LINE__, __PRETTY_FUNCTION__, ENDO_CYCLENOW())
+
+/* Private macros */
+
+#define ENDO_CYCLENOW() endoscope_cyclenow()
+#define ENDO_SYNCCLOCK(cycle, time) endoscope_syncclock(cycle, time)
+#define ENDO_GETTID() endoscope_gettid()
+#define ENDO_MUTEX_INIT(mu) \
```

Comment:
I used malloc so that mutex can be stored and passed in a void * pointer, and frontend (and the previous hack hole in the server class) doesn't need any implementation of mutex. This is no longer needed and is being removed.